webodf/lib/gui/TrivialUndoManager.js

```diff
@@ -294,6 +357,11 @@ gui.TrivialUndoManager = function TrivialUndoManager(defaultRules) {
             currentUndoState.push(op);
             eventNotifier.emit(gui.UndoManager.signalUndoStateModified, { operations: currentUndoState });
         }
+
+        newModified = isModified();
+        if (oldModified !== newModified) {
+            eventNotifier.emit(gui.UndoManager.signalModifiedChanged, newModified);
```

Comment:
This function is not invoked during an undo or redo operation. I suspect that performing an edit, clearing the modified state, then undoing the edit won't correctly mark the document as modified (unless my morning brain has missed something obvious here)